app/Http/Controllers/SquareController.php

```diff
@@ -0,0 +1,206 @@
+use App\Models\DuesTransaction;
+use App\Models\Payment;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Square\Models\CreateCheckoutRequest;
+use Square\Models\CreateOrderRequest;
+use Square\Models\Money;
+use Square\Models\Order;
+use Square\Models\OrderLineItem;
+use Square\Models\OrderServiceCharge;
+use Square\Models\OrderServiceChargeCalculationPhase;
+use Square\Models\OrderState;
+use Square\SquareClient;
+
+class SquareController extends Controller
+{
+    private const PER_TRANSACTION_FEE = 30;  // cents
+    private const PERCENTAGE_FEE = 2.9;
+
+    public function payDues(Request $request)
+    {
+        $user = $request->user();
+
+        $transactionWithNoPayment = DuesTransaction::doesntHave('payment')
+            ->where('user_id', $user->id)
+            ->latest('updated_at')
+            ->first();
+
+        $transactionWithIncompletePayment = DuesTransaction::where('user_id', $user->id)
+            ->whereHas('package', static function (Builder $q): void {
+                $q->whereDate('effective_end', '>=', date('Y-m-d'));
+            })->whereHas('payment', static function (Builder $q): void {
+                $q->where('amount', 0.00);
+                $q->where('method', 'square');
+            })->first();
+
+        if (null !== $transactionWithIncompletePayment) {
+            $transaction = $transactionWithIncompletePayment;
+
+            $payment = $transaction->payment[0];
+        } elseif (null !== $transactionWithNoPayment) {
+            $transaction = $transactionWithNoPayment;
+
+            $payment = new Payment();
+            $payment->amount = 0.00;
+            $payment->method = 'square';
+            $payment->recorded_by = $user->id;
+            $payment->unique_id = self::generateUniqueId();
+            $payment->notes = 'Checkout flow started';
+
+            $transaction->payment()->save($payment);
+        } else {
+            return view(
+                'square.error',
+                [
+                    'message' => 'We could not find a transaction ready for payment.',
+                ]
+            );
+        }
+
+        $amount = (int) ($transaction->package->cost * 100);
+
+        $basePrice = new Money();
+        $basePrice->setAmount($amount);
+        $basePrice->setCurrency('USD');
+
+        $surcharge = new Money();
+        $surcharge->setAmount(self::calculateSurcharge($amount));
+        $surcharge->setCurrency('USD');
+
+        $orderLineItem = new OrderLineItem('1');
+        $orderLineItem->setName('Dues - '.$transaction->package->name);
+        $orderLineItem->setBasePriceMoney($basePrice);
+
+        $orderServiceCharge = new OrderServiceCharge();
+        $orderServiceCharge->setName('Card Processing Surcharge');
+        $orderServiceCharge->setAmountMoney($surcharge);
+        $orderServiceCharge->setCalculationPhase(OrderServiceChargeCalculationPhase::TOTAL_PHASE);
```

Comment:
As noted in Slack, this is not exposed in the checkout UX super well. We could switch it to a tax but I think this is more semantically correct and is easier to calculate accurately.

Reply:
I did, no difference.

Reply:
Okay, that's fine then.